syntaxes/haskell.YAML-tmLanguage

```diff
@@ -423,15 +436,32 @@ patterns:
   - include: '#qualifier'
   - include: '#comments'
   - include: '#infix_op'
-  - include: '#double_colon'
   - begin: '(::|∷)(?![\p{S}\p{P}&&[^(),;\[\]`{}_"'']])'
     beginCaptures:
       '1': {name: keyword.operator.double-colon.haskell}
     end: >-
-      (?=\)|$|,|}|\b(in|then|else|of)\b(?!')|<\-)
+      (?x)
+        # End type annotation when seeing one of:
+        (?=
+          \)                          # closing parenthesis
+          |$                          # end of line
+          |,                          # comma
+          |}                          # closing bracket
+          |\b(in|then|else|of)\b(?!') # keyword
+          |                           # symbolic keyword except (->)
+          (?<![\p{S}\p{P}&&[^(),;\[\]`{}_"']])((<-)|(=)|(-<)|(-<<))([(),;\[\]`{}_"']|[^\p{S}\p{P}])
+        )
```

Comment:
Test for symbolic operators to end a type annotation.

Reply:
Very detailed commenting :smile: